fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java

```diff
@@ -408,7 +408,7 @@ public DiscreteValue(ExpressionRewriteContext context,
         public DiscreteValue(ExpressionRewriteContext context,
                 Expression reference, Expression toExpr, Collection<Literal> values) {
             super(context, reference, toExpr);
-            this.values = Sets.newTreeSet(values);
+            this.values = Sets.newHashSet(values);
```

Comment:
may cause options of `in` is no order.

check test on a in [1, 10, 20] union a in [100, 200] => a in [1, 10, 20, 100, 200]